### ISSUE TITLE
Why memory usage is growing, increasing from 17M to 40M.

### ISSUE THREAD

**Issue by zxl777** [realm/realm-cocoa#1749]
Why memory usage is growing, increasing from 17M to 40M.
The newest Realm SDK today. version 0.91.1

Source --

```objc
self.array = [MyRealObj allObjectsInRealm:EmojiRealm];

-(UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath
{
        static NSString * CellIdentifier = @"MyCell";
        MyCell * cell = [collectionView dequeueReusableCellWithReuseIdentifier:CellIdentifier forIndexPath:indexPath];
            MyRealObj * Obj = self.array[indexPath.row];                    
            cell.MyLable.text = obj.text ; 
            // cell.MyLable.text = @"xxx"; if using this line ,no memory problem.
            return cell;
 }
```

**Comment by segiddins** [comment]
@zxl777 the only thing that comes to mind looking at that code snippet is that fact that `@"xxx"` is a constant string, and therefore is never even allocated, whereas the text from the object will need to be allocated for every cell. If, for some reason, you have a huge amount of text in that property, or you're showing many cells at a time (or they're not being properly reused), that could explain some of the increase in memory usage. If it's neither of those things, however, we'd need an example app we could run to be able to instrument things. Hope that helps!

**Comment by zxl777** [comment]
I write a small app by these code ,no memory problem.
I think realm maybe have some memory bug in iOS8 Keyboard extension.

I have no time to solve this memory problem, So i use the plist file to replace realm.

But i think Realm is great.
thank you.

**Comment by segiddins** [comment]
@zxl777 if you could send us the version of your project that was using Realm, I'd really like to be able to dig in and see what me might be doing wrong. I'm sorry you ran into issues using Realm.

**Comment by zxl777** [comment]
Realm Version 0.91.1
I just run the keyboard in my app, memory is the same surge.
This experiment just denied my judgment.
This may not be Realm of the bug, but I really do not know what I did wrong.

I guess, Realm does not release the allocated memory.
I have confirmed that cell will be automatically released.

I can try to match debugging.
my iMessage: itoytoy@me.com

![2015-04-14 9 22 01](https://cloud.githubusercontent.com/assets/816448/7128644/c53cf67a-e287-11e4-8236-40348f76ce4f.png)

**Comment by segiddins** [comment]
@zxl777 again, if you want us to have a look, you can send a copy of your project to help@realm.io.

**Comment by zxl777** [comment]
Mail has been sent, please check.

**Comment by tgoyne** [comment]
The growth appears to be almost entirely due to loading the images for the emoji themselves, which does not happen when using a hardcoded string.

I measured this by launching the app, waiting for memory usage to stabilize, and then attaching Instruments to the process, measuring memory allocations and leaks. I then scrolled through the pages of the keyboard until the memory usage had increased by 14 MB. Instruments reports that 9.31 MB of the increase was `ImageIO_PNG_Data` and 1.65 MB was `VM: CoreAnimation`, leaving just 3 MB for everything else.
